### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Sudhanshusoni45**
### Implemneted Authentication
Login via Test Login to review.

[Login](https://4s7lni.csb.app/login)
[Signup](https://4s7lni.csb.app/signup)

### CODE REVIEW

**Review by Kevin-Solomon — APPROVED** [review]
Code looks good, but you try storing user details as well in the ```authState```

**Comment by Sudhanshusoni45** [comment]
> Code looks good, but you try storing user details as well in the `authState`

Yes thank you kevin , i also realised i've missed this after raising the pr.

File: src/context/auth-Context.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    try {
+      const url = "/api/auth/login";
+      const data = user;
+      const response = await axios.post(url, data);
+      console.log("response:", response);
+      const { encodedToken: token } = response.data;
+      if (response.status === 200) {
+        authDispatch({ type: "LOGIN", payload: { token: token } });
+        localStorage.setItem("token", token);
+        navigate("/home");
```
Comment (gadgetfather) [inline src/context/auth-Context.js]: Good job, checking the status before storing it.

**Review by gadgetfather — APPROVED** [review]
- Code is clean 
- Functionality also working fine

**Pull request merged**